### ISSUE TITLE
AOTI generated .so models output incorrect values

### ISSUE THREAD

**Issue by HDCharles** [pytorch/torchchat#294]
running eval/generate after exporting to .so vs not give verrry different (and unintelligible in the case of the .so) results.

# Repro
############################################### 
export MODEL_REPO=meta-llama/Llama-2-7b-hf
python export.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth \
    --dtype bfloat16 --device cuda \
    --output-dso-path checkpoints/$MODEL_REPO/model.so
python generate.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth \
    --dtype bfloat16 --device cuda \
    --dso-path checkpoints/$MODEL_REPO/model.so
python generate.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth \
    --dtype bfloat16 --device cuda
#########################################################

# Result

two python export.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth --dtype bfloat16 --device cuda --output-dso-path checkpoints/$MODEL_REPO/model.so

Warning: command export does not support option pte-path
Warning: command export does not support option dso-path
Namespace(seed=1234, prompt='Hello, my name is', tiktoken=False, chat=False, is_chat_model=False, gui=False, num_samples=1, max_new_tokens=200, top_k=200, temperature=0.8, compile=False, compile_prefill=False, profile=None, speculate_k=5, draft_checkpoint_path=None, checkpoint_path=PosixPath('checkpoints/meta-llama/Llama-2-7b-hf/model.pth'), checkpoint_dir=None, params_path=None, gguf_path=None, tokenizer_path=None, output_pte_path=None, output_dso_path='checkpoints/meta-llama/Llama-2-7b-hf/model.so', dso_path=None, pte_path=None, dtype='bfloat16', verbose=False, quantize='{ }', params_table=None, device='cuda', tasks=['wikitext'], limit=None, max_seq_length=None)
Using device=cuda
Loading model ...
name Llama-2-7b-hf
Time to load model: 11.23 seconds
Time to quantize model: 0.00 seconds
Exporting model using AOT Inductor to /home/cdhernandez/local/torchchat/checkpoints/meta-llama/Llama-2-7b-hf/model.so
len(input)=2
The generated DSO model can be found at: /home/cdhernandez/local/torchchat/checkpoints/meta-llama/Llama-2-7b-hf/model.so
#################################################
python generate.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth \
    --dtype bfloat16 --device cuda \
    --dso-path checkpoints/$MODEL_REPO/model.so
Warning: command generate does not support option output-pte-path
Warning: command generate does not support option output-dso-path
Warning: checkpoint path ignored because an exported DSO or PTE path specified
Warning: checkpoint path ignored because an exported DSO or PTE path specified
Loading model ...
name Llama-2-7b-hf
Time to load model: 11.80 seconds
/home/cdhernandez/local/pytorch/torch/backends/cuda/__init__.py:342: FutureWarning: torch.backends.cuda.sdp_kernel() is deprecated. In the future, this context manager will be removed. Please see, torch.nn.attention.sdpa_kernel() for the new context manager, with updated signature.
  warnings.warn(
Hello, my name is an an an, we have have is keeping a aka is the people, 14T 3 may have the same, perhaps the every risk taking I and the 2 certain he Daniel T: it =L0man an long T: Trans 1 when'good became, so as Take Care Principal mayer’looking in this’Future 4.
n...g thefian (the is a.
l8, we cour (O looked en.
 (U 91 99f784ne Re, maybe we any FY CH3, no200da–i/l3s (blue in 6-Work, sometimes run” to 200-turn. M in An/l3 and 4 will (asenthal.patchd their committee of the cannab, u22000da - a smell-R (R, told. TheWell?meb
###########################################################
python generate.py --checkpoint-path checkpoints/$MODEL_REPO/model.pth \
    --dtype bfloat16 --device cuda \

Warning: command generate does not support option output-pte-path
Warning: command generate does not support option output-dso-path
Loading model ...
name Llama-2-7b-hf
Time to load model: 10.79 seconds
Time to quantize model: 0.00 seconds
/home/cdhernandez/local/pytorch/torch/backends/cuda/__init__.py:342: FutureWarning: torch.backends.cuda.sdp_kernel() is deprecated. In the future, this context manager will be removed. Please see, torch.nn.attention.sdpa_kernel() for the new context manager, with updated signature.
  warnings.warn(
Hello, my name is TJ.
I am a stay at home dad with two kids under the age of 3. I am just starting to dabble in the world of 3D printing and I am hoping to learn as much as possible from the community. I have designed and printed a few things so far and I am hoping to share some of my designs in the near future.
Welcome to the forum. If you're in the UK we should get along swimmingly. What area are you in?
Thanks for the warm welcome! I'm in the Finger Lakes, NY about 2 hours from Rochester.
I'm in Middlesbrough, near the North Sea in the UK. MATCO Electronics has developed the next generation of the MATCO Automated Test Stand. Designed with the latest technology, it includes a powerful Intel Core i3 CPU and 500GB Solid State Drive.
